### PR TITLE
feat: react 18

### PR DESCRIPTION
**Pull request by QuiiBz**
## Type

- Migration

#### What is expected?

Upgrade to react(-dom) 18, react(-dom) types, scaleway/jest-helpers, testing-lib/react (to use the new createRoot API so tests are running in react 18).

This is based on #1314

### CODE REVIEW

**Review by philibea — APPROVED** [review]

File: src/components/List/variantExplorer.tsx [hunk 1]
```diff
@@ -127,8 +131,8 @@ export const Header = () => {
           aria-label={`sort ${label ?? index}`}
           tabIndex={label ? 0 : undefined}
           disabled={isLoading}
-          onClick={e => (label ? onSortEvent(e, sort, index) : undefined)}
-          onKeyPress={e => onSortEvent(e, sort, index)}
+          onClick={e => (label ? onSortEvent(e, index, sort) : undefined)}
+          onKeyPress={e => onSortEvent(e, index, sort)}
```
Comment (matthprost) [inline src/components/List/variantExplorer.tsx:135]:
```suggestion
          onClick={event => (label ? onSortEvent(event, index, sort) : undefined)}
          onKeyPress={event => onSortEvent(event, index, sort)}
```

by the way haha

File: src/components/List/variantProduct.tsx [hunk 2]
```diff
@@ -268,8 +272,8 @@ export const Header = () => {
           tabIndex={label ? 0 : undefined}
           aria-label={`sort ${label ?? index}`}
           disabled={isLoading}
-          onClick={e => (label ? onSortEvent(e, sort, index) : undefined)}
-          onKeyPress={e => onSortEvent(e, sort, index)}
+          onClick={e => (label ? onSortEvent(e, index, sort) : undefined)}
+          onKeyPress={e => onSortEvent(e, index, sort)}
```
Comment (matthprost) [inline src/components/List/variantProduct.tsx:276]:
```suggestion
          onClick={event => (label ? onSortEvent(event, index, sort) : undefined)}
          onKeyPress={event => onSortEvent(event, index, sort)}
```

File: src/components/List/variantTable.tsx [hunk 3]
```diff
@@ -171,8 +175,8 @@ export const Header = () => {
           tabIndex={label ? 0 : undefined}
           aria-label={`sort ${label ?? index}`}
           disabled={isLoading}
-          onClick={e => (label ? onSortEvent(e, sort, index) : undefined)}
-          onKeyPress={e => onSortEvent(e, sort, index)}
+          onClick={e => (label ? onSortEvent(e, index, sort) : undefined)}
+          onKeyPress={e => onSortEvent(e, index, sort)}
```
Comment (matthprost) [inline src/components/List/variantTable.tsx:179]:
```suggestion
          onClick={event => (label ? onSortEvent(event, index, sort) : undefined)}
          onKeyPress={event => onSortEvent(event, index, sort)}
```

File: src/components/Tooltip/index.tsx [hunk 4]
```diff
@@ -269,7 +280,11 @@ const Tooltip = ({
   }, [children, className, generatedId, onMouseEvent])
 
   if (!text) {
-    return children as JSX.Element
+    if (typeof children === 'function') {
+      return null
+    }
+
+    return <>{children}</>
```
Comment (matthprost) [inline src/components/Tooltip/index.tsx:287]:
```suggestion
    if (typeof children === 'function') return null

    return <>{children}</>
```

**Review by matthprost — CHANGES_REQUESTED** [review]

**Review by matthprost — APPROVED** [review]

**Pull request merged**